### PR TITLE
Moar refactoring!

### PR DESCRIPTION
**Pull request by davidfowl**
- Moved TraceSourceLogger to Microsoft.Framework.Logging.TraceSource
- Removed Logging dependency from all providers, they only need
the .Interfaces reference
- Moved WriteContext to the test project
- Added an extension method AddTraceSource to add the TraceSource provider
- Added AddLogging to the Logging project
- Removed RootTraceName

#142
#141
#140

### CODE REVIEW

File: src/Microsoft.Framework.Logging.TraceSource/TraceSourceFactoryExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
```
Comment (Eilon) [inline src/Microsoft.Framework.Logging.TraceSource/TraceSourceFactoryExtensions.cs]: Headerz

Comment (davidfowl) [inline src/Microsoft.Framework.Logging.TraceSource/TraceSourceFactoryExtensions.cs]: Do we have a script for that?

Comment (Eilon) [inline src/Microsoft.Framework.Logging.TraceSource/TraceSourceFactoryExtensions.cs]: No but I'll go ahead and do it.

Comment (Eilon) [inline src/Microsoft.Framework.Logging.TraceSource/TraceSourceFactoryExtensions.cs]: Fixed in https://github.com/aspnet/Logging/pull/145

**Pull request merged**